hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -256,6 +260,7 @@ private void handlePlatformTransaction(
 
             // Now that we have a created handle context object and a consensus timestamp, run the appropriate {@code
             // ConsensusTimeHook} event handlers
+            genesisRecordsTimeHook.process(consensusNow, context);
```

Comment:
This feels so wrong. We must call it once during Genesis, i.e., never in production. But it is handled for each transaction. I wonder if there is a better way. I suggest writing the check in a super fast way. Maybe we can put this into simple `Consumer` that takes the `consensusNow` and is `null` if this method does not need to be called. This would reduce the code in production to a simple `null`-check.